clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -916,7 +917,9 @@ struct ConvertCIRToLLVMPass
   }
   void runOnOperation() final;
 
-  virtual StringRef getArgument() const override { return "cir-to-llvm"; }
+  virtual StringRef getArgument() const override {
+    return "cir-to-llvm-internal";
```

Comment:
what does `internal` means? I don't understand why this needs to be renamed.

Reply:
>  I would deprecate createConvertCIRToLLVMPass of even remove it from the interface, but it;s up to you to decide.

I'm down for simplifying and unifying as much as possible, but can you address it in a follow up PR.

> internal here is a hint that this pass can't be run with any CIR input like it was before. Now the pre requirement is our new cir-flatten-cfg pass

Perhaps it should be named `cir-flat-to-llvm`, or something like that.

Since it would be wrong to run `cir-flat-to-llvm` without running `cir-flatten-cfg`, we should make `cir-flat-to-llvm` pass to require `cir-flatten-cfg` pass to be run before.

Seems like all the changes for this comment should be addressed together in another PR.